transform/mattermost-analytics/models/marts/product/_product__models.yml

```diff
@@ -240,3 +221,19 @@ models:
       - name: is_missing_activity_data
         description: Whether server activity data are available for the given date and server id.
 
+  - name: dim_server_info
+    description: Static information for a given server.
+    columns:
+      - name: server_id
+        description: The server's unique id.
+        tests:
+          - not_null
+      - name: installation_id
+        description: The id of the installation. Exists only for cloud installations.
+      - name: hosting_type
+        description: |
+          Type of hosting for this server. Can be one of:
+          - `Cloud`: server always reports an installation id
+          - `Self-hosted`: server always reports a null installation id.
+          - `Probably cloud`: server reports an installation id more often than reporting null as installation id.
```

Comment:
for 'Probably Cloud' you compare against `count_not_is_cloud_days` not against `count_unknown_is_cloud_days` (which covers the nulls). Same for `Probably self-hosted`.

Reply:
It's a bit tricky. In the date spine per server id table, a row exists for each combination of server id and every date between the first and last date any telemetry/security update exists. Unfortunately there is no guarantee that data exists for each date. For example, a server might have sent it's first telemetry on `2023-08-01` and it's last on `2023-08-07`, but telemetry is missing on `2023-08-03`. It's those cases where `is_cloud` is null, as there is no data and it's impossible to check if the server has an installation id or not. 

The check for deciding if the server is cloud or not uses the following logic:
- Count the number of days where the server is reporting it's a cloud server.
- Count the number of days where the server is reporting it's a self-hosted server.
- If the server reports only cloud server days, then it's definitely a cloud server.
- If the server reports only self-hosted server days, then it's definitely a self-hosted server.
- If the server reports both self-hosted and cloud, use the largest value of the two, but mark it as `Probably`.
- If there's no information regarding installation id in activity data, it's impossible to decide if it's a cloud or on prem server based on installation id. A more elaborate solution should be added.

A few final remarks:
- `count_unknown_is_cloud_days` is not used. Perhaps it should be removed for avoiding confusion.
- The majority of the servers reports either cloud or self hosted days. There are only ~ 43 servers that report both, and should be investigated as special cases.

Reply:
Thanks for the detailed answer. Made the comment because you mentioned in the description: "server reports an installation id more often than reporting null" (and I thought that `count_unknown_is_cloud_days ` captures this semantics) as opposed to "server reports both self-hosted and cloud, use the largest value of the two". Agree that better not have `count_unknown_is_cloud_days` if not used (or replace it with a TODO statement instead)

Reply:
Removed.